--- a/tk/valoeghese/fc0/client/Main.java
+++ b/tk/valoeghese/fc0/client/Main.java
@@ -7,9 +7,9 @@
 
 public class Main {
-    public static void main(String[] object) {
-        object = "Starting 2fc0f18 Client.";
+    public static void main(String[] args) {
+        Object object = "Starting 2fc0f18 Client.";
         System.out.println(object);
         object = new Fc0();
-        object.run();
+        ((Fc0) object).run();
         object = "Client shut down.";
         System.out.println(object);
