--- a/tk/valoeghese/fc0/client/Player.java
+++ b/tk/valoeghese/fc0/client/Player.java
@@ -16,4 +16,7 @@
 import tk.valoeghese.fc0.world.TileView;
 
+import java.util.function.IntFunction;
+import java.util.function.UnaryOperator;
+
 public final class Player {
     private final MutableVec3d pos = new MutableVec3d(0.0, 0.0, 0.0);
@@ -116,8 +119,8 @@
         double d15 = d5;
         double d16 = d4;
-        Object object = d3 < Math.PI ? Direction.EAST : Direction.WEST;
+        Direction direction1 = d4 < Math.PI ? Direction.EAST : Direction.WEST;
         Direction direction2 = d15 > 0.0 ? Direction.DOWN : Direction.UP;
         Direction direction3 = d16 > var_double_a && d16 < var_double_b ? Direction.SOUTH : Direction.NORTH;
-        object = arg_0 -> Player.a((Direction)object, direction2, direction3, arg_0);
+        IntFunction<Direction> object = arg_0 -> Player.a((Direction)direction1, direction2, direction3, arg_0);
         do {
             double d17 = -d9 / Math.tan(d4);
