--- a/tk/valoeghese/fc0/client/Player.java
+++ b/tk/valoeghese/fc0/client/Player.java
@@ -116,8 +116,8 @@
         double d15 = d5;
         double d16 = d4;
-        Object object = d3 < Math.PI ? Direction.EAST : Direction.WEST;
+        Object direction1 = d4 < Math.PI ? Direction.EAST : Direction.WEST;
         Direction direction2 = d15 > 0.0 ? Direction.DOWN : Direction.UP;
         Direction direction3 = d16 > field_116 && d16 < field_123 ? Direction.SOUTH : Direction.NORTH;
-        object = arg_0 -> Player.method_94((Direction)object, direction2, direction3, arg_0);
+        java.util.function.IntFunction<Direction> object = arg_0 -> Player.method_94((Direction)direction1, direction2, direction3, arg_0);
         do {
             double d17 = -d9 / Math.tan(d4);
