--- a/tk\valoeghese\fc0\client\render\OtherTesselator.java
+++ b/tk\valoeghese\fc0\client\render\OtherTesselator.java
@@ -19,5 +19,5 @@
     private int var_int_a = 0;
     private IntList var_it_unimi_dsi_fastutil_ints_IntList_a = new IntArrayList();
-    private List var_java_util_List_a = new ArrayList();
+    private List<OtherVertex> var_java_util_List_a = new ArrayList<>();
     private final int b;
     private final int c;
@@ -60,5 +60,5 @@
         GL20.glEnableVertexAttribArray(1);
         GL30.glBindVertexArray(0);
-        this.var_java_util_List_a.add(new OtherVertex(n3, arrn.length, 0));
+        this.var_java_util_List_a.add(new OtherVertex(n3, arrn.length, (byte) 0));
     }
 
