--- a/tk/valoeghese/fc0/client/render/TerrainRenderer.java
+++ b/tk/valoeghese/fc0/client/render/TerrainRenderer.java
@@ -15,7 +15,7 @@
     public TerrainRenderer(List<TileRenderer> object) {
         super(35048, Player.field_172);
-        object = object.iterator();
-        while (object.hasNext()) {
-            TileRenderer tileRenderer = (TileRenderer)object.next();
+        java.util.Iterator<TileRenderer> iter = object.iterator();
+        while (iter.hasNext()) {
+            TileRenderer tileRenderer = iter.next();
             tileRenderer.apply(this);
         }
