--- a/tk/valoeghese/fc0/client/render/Textures.java
+++ b/tk/valoeghese/fc0/client/render/Textures.java
@@ -20,10 +20,9 @@
         }
         catch (IOException iOException) {
-            object = iOException;
             iOException.printStackTrace();
             if (bl) {
                 return 0;
             }
-            throw new UncheckedIOException("Error loading Tile Atlas", (IOException)object);
+            throw new UncheckedIOException("Error loading Tile Atlas", (IOException)iOException);
         }
     }
