--- a/tk\valoeghese\fc0\client\render\WorldRenderer.java
+++ b/tk\valoeghese\fc0\client\render\WorldRenderer.java
@@ -21,5 +21,5 @@
     private int vertexCount = 0;
     private IntList indices = new IntArrayList();
-    private List vertices = new ArrayList();
+    private List<WorldVertex> vertices = new ArrayList<>();
     private final int usage;
     private final Shader shader;
@@ -66,5 +66,5 @@
         GL20.glEnableVertexAttribArray(2);
         GL30.glBindVertexArray(0);
-        this.vertices.add(new WorldVertex(n3, arrn.length, 0));
+        this.vertices.add(new WorldVertex(n3, arrn.length, (byte) 0));
     }
 
