--- a/tk/valoeghese/fc0/world/World.java
+++ b/tk/valoeghese/fc0/world/World.java
@@ -40,5 +40,5 @@
         this.chunks = new Chunk[this.chunkSize * this.chunkSize];
         this.var_java_util_Random_a = new Random(l);
-        OrderedList orderedList = new OrderedList(chunk -> Math.abs(chunk.width) + Math.abs(chunk.depth));
+        OrderedList<Chunk> orderedList = new OrderedList<>(chunk -> Math.abs(chunk.width) + Math.abs(chunk.depth));
         for (n = -8; n < 9; ++n) {
             for (int i = -8; i < 9; ++i) {
@@ -85,5 +85,5 @@
     }
 
-    public final List a() {
+    public final List<Chunk> a() {
         if (!this.var_java_util_Queue_a.isEmpty()) {
             boolean bl = this.var_boolean_a = !this.var_boolean_a;
