--- a/tk/valoeghese/fc0/world/generator/TreeGenerator.java
+++ b/tk/valoeghese/fc0/world/generator/TreeGenerator.java
@@ -16,5 +16,5 @@
     }
 
-    public void method_321(TileView tileView, TreeGeneratorConfig treeGeneratorConfig, int n, int n2, Random random) {
+    public void generate(TileView tileView, TreeGeneratorConfig treeGeneratorConfig, int n, int n2, Random random) {
         int n3 = treeGeneratorConfig.getTreeGenAmount(n, n2);
         if (random.nextInt(8) == 0) {
