--- a/tk/valoeghese/fc0/util/Direction.java
+++ b/tk/valoeghese/fc0/util/Direction.java
@@ -7,13 +7,12 @@
 import tk.valoeghese.fc0.math.TilePos;
 
-public final class Direction
-extends Enum<Direction>
+public enum Direction
 implements UnaryOperator<TilePos> {
-    public static final /* enum */ Direction NORTH = new Direction(0, 0, 1);
-    public static final /* enum */ Direction SOUTH = new Direction(0, 0, -1);
-    public static final /* enum */ Direction EAST = new Direction(1, 0, 0);
-    public static final /* enum */ Direction WEST = new Direction(-1, 0, 0);
-    public static final /* enum */ Direction UP = new Direction(0, 1, 0);
-    public static final /* enum */ Direction DOWN = new Direction(0, -1, 0);
+    NORTH(0, 0, 1),
+    SOUTH(0, 0, -1),
+    EAST(1, 0, 0),
+    WEST(-1, 0, 0),
+    UP(0, 1, 0),
+    DOWN(0, -1, 0);
     private final int x;
     private final int y;
@@ -44,5 +43,5 @@
 
     @Override
-    public /* synthetic */ Object apply(Object object) {
+    public /* synthetic */ TilePos apply(TilePos object) {
         return this.method_85((TilePos)object);
     }
