--- a/tk/valoeghese/fc0/client/gui/GuiVertex.java
+++ b/tk/valoeghese/fc0/client/gui/GuiVertex.java
@@ -45,5 +45,5 @@
 
     public static void initAudio() {
-        audioDevice = ALC10.alcOpenDevice(null);
+        audioDevice = ALC10.alcOpenDevice((ByteBuffer) null);
         ALCCapabilities aLCCapabilities = ALC.createCapabilities(audioDevice);
         IntBuffer intBuffer = BufferUtils.createIntBuffer(16);
