--- a/tk/valoeghese/fc0/client/gui/I18n.java
+++ b/tk/valoeghese/fc0/client/gui/I18n.java
@@ -18,7 +18,7 @@
     private final String parent;
     private Properties translations;
-    public static final I18n INSTANCE = new I18n("en.gb", "British English");
+    public static final I18n INSTANCE;
 
-    private I18n(String string, String string2) {
+    private I18n(String string, String string2) throws Throwable {
         this(string, string2, null);
     }
@@ -29,9 +29,12 @@
      * Enabled aggressive exception aggregation
      */
-    private I18n(String object, String object2, @Nullable String object3) {
+    private I18n(String id, String name, @Nullable String parent) throws Throwable {
         Throwable throwable;
+        Object object;
+        Object object2;
+        Object object3;
         block19: {
-            this.id = object;
-            this.parent = object3;
+            this.id = id;
+            this.parent = parent;
             object = this;
             object2 = null;
@@ -49,5 +52,5 @@
                             object = throwable2;
                             throwable = throwable2;
-                            throw object;
+                            throw throwable2;
                         }
                         catch (Throwable throwable3) {
@@ -86,5 +89,5 @@
                     object = throwable5;
                     throwable = throwable5;
-                    throw object;
+                    throw throwable5;
                 }
                 catch (Throwable throwable6) {
@@ -118,5 +121,10 @@
 
     static {
-        new I18n("en.us", "American English", "en.gb");
+        try {
+            INSTANCE = new I18n("en.gb", "British English");
+            new I18n("en.us", "American English", "en.gb");
+        } catch (Throwable t) {
+            throw new RuntimeException(t);
+        }
     }
 }
