--- a/tk/valoeghese/fc0/math/TilePos.java
+++ b/tk/valoeghese/fc0/math/TilePos.java
@@ -19,10 +19,5 @@
 
     public TilePos(Pos pos) {
-        double d = pos.x;
-        double d2 = d;
-        double d3 = pos.y;
-        d2 = d3;
-        d2 = pos.z;
-        this((int)Math.floor(d), (int)Math.floor(d3), (int)Math.floor(d2));
+        this((int)Math.floor(pos.x), (int)Math.floor(pos.y), (int)Math.floor(pos.z));
     }
 
