--- a/tk/valoeghese/fc0/math/TilePos.java
+++ b/tk/valoeghese/fc0/math/TilePos.java
@@ -18,10 +18,5 @@
 
     public TilePos(Pos pos) {
-        double d = pos.x + 0.5;
-        double d2 = d;
-        double d3 = pos.y + 0.5;
-        d2 = d3;
-        d2 = pos.z + 0.5;
-        this((int)Math.floor(d), (int)Math.floor(d3), (int)Math.floor(d2));
+        this((int)Math.floor(pos.x + 0.5), (int)Math.floor(pos.y + 0.5), (int)Math.floor(pos.z + 0.5));
     }
 
