--- a/tk/valoeghese/fc0/io/ResourceLoader.java
+++ b/tk/valoeghese/fc0/io/ResourceLoader.java
@@ -18,7 +18,7 @@
     }
 
-    public static String read(String object) {
+    public static String read(String location) throws java.io.IOException {
         int n;
-        object = ResourceLoader.load((String)object);
+        Object object = ResourceLoader.load((String)location);
         ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
         byte[] arrby = new byte[16384];
