--- a/tk/valoeghese/fc0/io/ThreadedSodChunkStorage.java
+++ b/tk/valoeghese/fc0/io/ThreadedSodChunkStorage.java
@@ -71,6 +71,5 @@
      */
     public void saveChunks(Iterator<? extends Chunk> iterator) {
-        Object object = LOCK;
-        synchronized (object) {
+        synchronized (LOCK) {
             try {
                 while (saveThread != null && !SaveThread.isReady()) {
@@ -108,6 +107,5 @@
      */
     public void save(Iterator<? extends Chunk> iterator, Iterator<Item> iterator2, int n, Pos pos, Pos pos2, long l) {
-        Object object = LOCK;
-        synchronized (object) {
+        synchronized (LOCK) {
             try {
                 while (saveThread != null && !SaveThread.isReady()) {
