--- a/tk/valoeghese/fc0/world/ChunkStatus.java
+++ b/tk/valoeghese/fc0/world/ChunkStatus.java
@@ -4,19 +4,6 @@
 package tk.valoeghese.fc0.world;
 
-public final class ChunkStatus
-extends Enum<ChunkStatus> {
-    public static final /* enum */ ChunkStatus GENERATE = new ChunkStatus();
-    public static final /* enum */ ChunkStatus POPULATE = new ChunkStatus();
-    public static final /* enum */ ChunkStatus TICK = new ChunkStatus();
-    public static final /* enum */ ChunkStatus RENDER = new ChunkStatus();
-    private static final /* synthetic */ ChunkStatus[] _VALUES;
-
-    public static ChunkStatus[] values() {
-        return (ChunkStatus[])_VALUES.clone();
-    }
-
-    static {
-        _VALUES = new ChunkStatus[]{GENERATE, POPULATE, TICK, RENDER};
-    }
+public enum ChunkStatus {
+    GENERATE, POPULATE, TICK, RENDER,
 }
 
