--- a/tk/valoeghese/fc0/world/Chunk.java
+++ b/tk/valoeghese/fc0/world/Chunk.java
@@ -55,7 +55,7 @@
             for (n2 = 0; n2 < 16; ++n2) {
                 for (int j = 0; j < 16; ++j) {
-                    object = Tile.BY_ID[this.getTile(n2, i, j)];
-                    this.field_146 += ((Tile)object).field_165;
-                    if (n == 0 || !((Tile)object).isNotAir()) continue;
+                    Tile t = Tile.BY_ID[this.getTile(n2, i, j)];
+                    this.field_146 += t.field_165;
+                    if (n == 0 || !t.isNotAir()) continue;
                     this.presentYLevels.add(i);
                     n = 0;
@@ -127,7 +127,6 @@
             ((Entity)object).chunk = this;
             Entity entity = object;
-            object = this;
-            if (!((Chunk)object).entities.contains(entity)) {
-                ((Chunk)object).entities.add(entity);
+            if (!entities.contains(entity)) {
+                entities.add(entity);
             }
         }
