--- a/tk/valoeghese/fc0/world/Chunk.java
+++ b/tk/valoeghese/fc0/world/Chunk.java
@@ -109,7 +109,6 @@
             ((Player)object).chunk = this;
             Player player = object;
-            object = this;
-            if (!((Chunk)object).players.contains(player)) {
-                ((Chunk)object).players.add(player);
+            if (!players.contains(player)) {
+                players.add(player);
             }
         }
