--- a/tk/valoeghese/fc0/world/World.java
+++ b/tk/valoeghese/fc0/world/World.java
@@ -44,7 +44,7 @@
             System.out.println("Generating World.");
         }
-        this.chunks = (Chunk[])intFunction.apply(this.chunkSize * this.chunkSize);
+        this.chunks = intFunction.apply(this.chunkSize * this.chunkSize);
         this.random = new Random(l);
-        this.orderedChunks = new OrderedList<Chunk>(chunk -> Math.abs(chunk.x) + Math.abs(chunk.z));
+        this.orderedChunks = new OrderedList<>(chunk -> Math.abs(chunk.x) + Math.abs(chunk.z));
         for (int i = -n + 1; i < n; ++i) {
             for (int j = -n + 1; j < n; ++j) {
