--- a/tk/valoeghese/fc0/client/world/ClientChunk.java
+++ b/tk/valoeghese/fc0/client/world/ClientChunk.java
@@ -28,10 +28,7 @@
         }
         super.setAndUpdateTile(n, n2, n3, by);
-        int n5 = n4;
-        n4 = by;
-        by = (byte)n5;
         ClientChunk clientChunk3 = this;
         if (clientChunk3.renderer != null) {
-            clientChunk3.renderer.setTile(by, (byte)n4);
+            clientChunk3.renderer.setTile(n4, by);
         }
         if (n == 0) {
