--- a/tk/valoeghese/sod/DoubleArrayDataSection.java
+++ b/tk/valoeghese/sod/DoubleArrayDataSection.java
@@ -1,35 +1,43 @@
-/*
- * Decompiled with CFR 0.150.
- */
 package tk.valoeghese.sod;
 
+import java.util.Iterator;
+
 import it.unimi.dsi.fastutil.doubles.DoubleArrayList;
 import it.unimi.dsi.fastutil.doubles.DoubleList;
-import java.util.Iterator;
-import tk.valoeghese.sod.BaseDataSection;
 
-public final class DoubleArrayDataSection
-implements BaseDataSection<Double> {
-    private final DoubleList array = new DoubleArrayList();
+public class DoubleArrayDataSection implements BaseDataSection<Double> {
+    public DoubleArrayDataSection() {
+        this.array = new DoubleArrayList();
+    }
 
-    public final int size() {
-        return this.array.size();
+    private final DoubleList array;
+
+    public void writeDouble(double value) {
+        this.array.add(value);
     }
 
+    public int size() {
+        return array.size();
+    }
+
+    /**
+     * @deprecated Should only be used by the parser! Please use the type specific methods instead for writing data.
+     */
     @Override
-    public final <T> void writeForParser(T object) {
-        if (object instanceof Double) {
-            double d = (Double)object;
-            object = this;
-            ((DoubleArrayDataSection)object).array.add(d);
-            return;
+    public <T> void writeForParser(T data) throws UnsupportedOperationException {
+        if (data instanceof Double) {
+            this.writeDouble((Double) data);
+        } else {
+            throw new UnsupportedOperationException("Invalid data type parameter for this data section");
         }
-        throw new UnsupportedOperationException("Invalid data type parameter for this data section");
     }
 
+    public double readDouble(int index) {
+        return this.array.getDouble(index);
+    }
+
     @Override
-    public final Iterator<Double> iterator() {
+    public Iterator<Double> iterator() {
         return this.array.iterator();
     }
 }
-
