--- a/tk/valoeghese/sod/FloatArrayDataSection.java
+++ b/tk/valoeghese/sod/FloatArrayDataSection.java
@@ -1,35 +1,43 @@
-/*
- * Decompiled with CFR 0.150.
- */
 package tk.valoeghese.sod;
 
+import java.util.Iterator;
+
 import it.unimi.dsi.fastutil.floats.FloatArrayList;
 import it.unimi.dsi.fastutil.floats.FloatList;
-import java.util.Iterator;
-import tk.valoeghese.sod.BaseDataSection;
 
-public final class FloatArrayDataSection
-implements BaseDataSection<Float> {
-    private final FloatList array = new FloatArrayList();
+public class FloatArrayDataSection implements BaseDataSection<Float> {
+    public FloatArrayDataSection() {
+        this.array = new FloatArrayList();
+    }
 
-    public final int size() {
-        return this.array.size();
+    private final FloatList array;
+
+    public void writeFloat(float value) {
+        this.array.add(value);
     }
 
+    public int size() {
+        return array.size();
+    }
+
+    /**
+     * @deprecated Should only be used by the parser! Please use the type specific methods instead for writing data.
+     */
     @Override
-    public final <T> void writeForParser(T object) {
-        if (object instanceof Float) {
-            float f = ((Float)object).floatValue();
-            object = this;
-            ((FloatArrayDataSection)object).array.add(f);
-            return;
+    public <T> void writeForParser(T data) throws UnsupportedOperationException {
+        if (data instanceof Float) {
+            this.writeFloat((Float) data);
+        } else {
+            throw new UnsupportedOperationException("Invalid data type parameter for this data section");
         }
-        throw new UnsupportedOperationException("Invalid data type parameter for this data section");
     }
 
+    public float readFloat(int index) {
+        return this.array.getFloat(index);
+    }
+
     @Override
-    public final Iterator<Float> iterator() {
+    public Iterator<Float> iterator() {
         return this.array.iterator();
     }
 }
-
