--- a/tk/valoeghese/sod/Parser.java
+++ b/tk/valoeghese/sod/Parser.java
@@ -22,5 +22,5 @@
 
 final class Parser {
-    static BinaryData parse(DataInputStream dataInputStream) {
+    static BinaryData parse(DataInputStream dataInputStream) throws java.io.IOException {
         BaseDataSection baseDataSection;
         DataType dataType;
@@ -205,8 +205,8 @@
     }
 
-    static void write(BinaryData binaryData, DataOutputStream dataOutputStream) {
+    static void write(BinaryData binaryData, DataOutputStream dataOutputStream) throws java.io.IOException {
         dataOutputStream.writeLong(10976542L);
         for (Map.Entry<String, BaseDataSection> entry : binaryData) {
-            Iterator<Object> iterator;
+            Iterator<?> iterator;
             BaseDataSection baseDataSection = entry.getValue();
             if (baseDataSection instanceof DataSection) {
