--- a/tk/valoeghese/fc0/client/input/KeyboardInputHandler.java
+++ b/tk/valoeghese/fc0/client/input/KeyboardInputHandler.java
@@ -10,8 +10,7 @@
 import tk.valoeghese.fc0.client.input.Keybind;
 
-public final class KeyboardInputHandler
-extends Enum<KeyboardInputHandler>
+public enum KeyboardInputHandler
 implements GLFWKeyCallbackI {
-    public static final /* enum */ KeyboardInputHandler INSTANCE;
+    INSTANCE;
     static final Int2ObjectMap<List<Keybind>> bindings;
 
@@ -19,5 +18,5 @@
     public final void invoke(long l, int n, int n2, int n3, int n4) {
         if (n <= 348 && bindings.containsKey(n)) {
-            for (Keybind keybind : (List)bindings.get(n)) {
+            for (Keybind keybind : bindings.get(n)) {
                 keybind.update(n3);
             }
@@ -26,5 +25,4 @@
 
     static {
-        (new KeyboardInputHandler[1])[0] = INSTANCE = new KeyboardInputHandler();
         bindings = new Int2ObjectArrayMap<List<Keybind>>();
     }
