--- a/tk/valoeghese/fc0/client/input/MouseInputHandler.java
+++ b/tk/valoeghese/fc0/client/input/MouseInputHandler.java
@@ -11,5 +11,4 @@
 
 public final class MouseInputHandler
-extends Enum<MouseInputHandler>
 implements GLFWMouseButtonCallbackI {
     public static final /* enum */ MouseInputHandler INSTANCE;
@@ -19,5 +18,5 @@
     public final void invoke(long l, int n, int n2, int n3) {
         if (n <= 7 && bindings.containsKey(n)) {
-            for (Keybind keybind : (List)bindings.get(n)) {
+            for (Keybind keybind : bindings.get(n)) {
                 keybind.update(n2);
             }
