--- a/tk/valoeghese/fc0/class_106.java
+++ b/tk/valoeghese/fc0/class_106.java
@@ -30,5 +30,5 @@
                 throw new RuntimeException("Invalidly sized image encountered while generating atlas \"" + string + "\"!");
             }
-            this.field_371.put(class_1072.field_373, new class_108(n, n2, 0));
+            this.field_371.put(class_1072.field_373, new class_108(n, n2));
             int n3 = n << 4;
             int n4 = 15 - n2 << 4;
