--- a/tk/valoeghese/fc0/class_84.java
+++ b/tk/valoeghese/fc0/class_84.java
@@ -45,7 +45,7 @@
     public final void method_256(int n, @Nullable Item object) {
         super.method_256(n, (Item)object);
-        object = this.field_294.method_289();
-        if (object != null) {
-            ((InventoryGui)object).method_278(n, this.field_294.method_285());
+        InventoryGui gui = this.field_294.method_289();
+        if (gui != null) {
+            gui.method_278(n, this.field_294.method_285());
         }
     }
